Server/src/kafka/MsgBusImpl_kafka.cpp

```diff
@@ -373,6 +373,11 @@ void msgBus_kafka::produce(const char *topic_var, char *msg, size_t msg_size, in
         sleep(1);
     }
 
+    // if topic is disabled, don't bother producing the message
+    // TODO: it would be more efficient to move this check to the top of the various update_* methods, but I'm not sure which parts of these methods have side-effects that need to be preserved.
```

Comment:
We can leave this for now, but what would be good is to do this check in the constructor and then update the appropriate  **update\_** methods check if they are enabled or not.  We can add that check post their required reference updates, such as hash updates.